File: data/solutions/P5/solution.py
num1 = input()
num2 = input()

def addStrings(num1: str, num2: str) -> str:
    res = []
    i, j = len(num1) - 1, len(num2) - 1
    carry = 0
    while i >= 0 or j >= 0 or carry:
        n1 = int(num1[i]) if i >= 0 else 0
        n2 = int(num2[j]) if j >= 0 else 0
        
        current_sum = n1 + n2 + carry
        carry = current_sum // 10
        res.append(str(current_sum % 10))
        
        i -= 1
        j -= 1
    return "".join(res[::-1])

print(addStrings(num1, num2))

File: data/solutions/P5/solution.cpp
#include <iostream>
#include <string>
#include <algorithm> // For std::reverse

std::string addStrings(std::string num1, std::string num2) {
    std::string res = "";
    int i = num1.length() - 1, j = num2.length() - 1, carry = 0;
    while (i >= 0 || j >= 0 || carry) {
        int n1 = (i >= 0) ? num1[i--] - '0' : 0;
        int n2 = (j >= 0) ? num2[j--] - '0' : 0;
        int sum = n1 + n2 + carry;
        res += std::to_string(sum % 10);
        carry = sum / 10;
    }
    std::reverse(res.begin(), res.end());
    return res;
}

int main() {
    std::string num1, num2;
    std::cin >> num1 >> num2;
    std::cout << addStrings(num1, num2) << std::endl;
    return 0;
}